waku/v2/api/publish/message_queue.go

```diff
@@ -0,0 +1,158 @@
+package publish
+
+import (
+	"container/heap"
+	"context"
+
+	"github.com/waku-org/go-waku/waku/v2/protocol"
+)
+
+// MessagePriority determines the ordering for the message priority queue
+type MessagePriority = int
+
+const (
+	LowPriority    MessagePriority = 1
+	MediumPriority MessagePriority = 2
+	HighPriority   MessagePriority = 3
+
+	DefaultPriority MessagePriority = MediumPriority
```

Comment:
Make LowPriority as the default may give more flexibility, as most of messages is default and low (maybe renamed it to NormalPriority and remove the const DefaultPriority). 
A few will be MediumPriority, and only occasionally there is HighPriority messages.

Reply:
I don't think having all the messages `LowPriority` by default is a good idea. Having the messages being by default a medium priority will give you more flexibility as some messages you probably don't care much for them to be sent immediately, while others you want them to be delivered ASAP. Having all the messages by default being Low would mean that you cannot deprioritize the messages any further.

However, renaming `MediumPriority` to `NormalPriority` does in fact sound like a good idea!